src/server/util/aws.ts

```diff
@@ -10,48 +11,39 @@ export enum FileVisibility {
 
 export const s3 = new S3()
 
-/**
- * Reformat the pre-signed url to one that will be accepted by S3.
- *
- * This is necessary because we used a gov.sg CNAME and configured S3
- * to only accept requests that have that gov.sg CNAME as the path.
- *
- * Broadly speaking, we need to change https://s3.amazonaws.com/file.go.gov.sg/link?search
- * to https://file.go.gov.sg.s3.amazonaws.com/link?search.
- */
-const reformatPresignedUrl = (url: string, fileName: string) => {
-  const urlObj = parse(url)
-  const { host, pathname, protocol, search } = urlObj
-  const newUrl = new URL('https://lorem-ipsum.com')
-  newUrl.protocol = protocol as string
-  newUrl.host = `${pathname?.split('/')[1]}.${host}`
-  newUrl.pathname = fileName
-  newUrl.search = search as string
-  return newUrl.href
+export interface S3Interface {
+  setS3ObjectACL: (key: string, acl: FileVisibility) => Promise<any>
+  uploadFileToS3: (file: Buffer, key: string, fileType: string) => Promise<any>
+  isValidS3Shortlink: (url: string, shortUrl: string) => boolean
 }
 
-export const generatePresignedUrl = async (
-  fileName: string,
-  fileType: string,
-) => {
-  const params = {
-    Bucket: s3Bucket,
-    Key: fileName,
-    ContentType: fileType,
-    Expires: 60, // 60 seconds.
+@injectable()
+/* eslint class-methods-use-this: ["error", { "exceptMethods":
+  ["setS3ObjectACL", "uploadFileToS3", "isValidS3Shortlink"] }] */
```

Comment:
The methods in the class can be declared _static_, then there would be no need to circumvent ESLint

Reply:
Looks like using the `static` keyword would cause the typescript error where the method is not recognised as one that would fulfill the `implements` keyword. [This is because when a class implements an interface, only the instance side of the class is checked.](https://www.typescriptlang.org/docs/handbook/interfaces.html#difference-between-the-static-and-instance-sides-of-classes)